pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -384,18 +383,10 @@ def __validate_init_configuration(self) -> None:
                     ' will duplicate the last checkpoint saved.'
                 )
 
-    def __init_ckpt_dir(
-        self,
-        dirpath: Optional[Union[str, Path]],
-        filename: Optional[str],
-        save_top_k: Optional[int],
-    ) -> None:
+    def __init_ckpt_dir(self, dirpath: Optional[Union[str, Path]], filename: Optional[str]) -> None:
         self._fs = get_filesystem(str(dirpath) if dirpath else '')
 
-        if (
-            save_top_k is not None and save_top_k > 0 and dirpath is not None and self._fs.isdir(dirpath)
-            and len(self._fs.ls(dirpath)) > 0
-        ):
+        if self.save_top_k != 0 and dirpath is not None and self._fs.isdir(dirpath) and len(self._fs.ls(dirpath)) > 0:
```

Comment:
careful, are you sure?
previously we had 

`save_top_k is not None and save_top_k > 0 and ...
`
and now we would have just

`save_top_k > 0 and ...
`
but you included -1? Double check?

Reply:
It does not match exactly the previous logic but I think it was wrong.

This check is to warn that the checkpoint directory is not empty.
So we should warn whenever a checkpoint might get created
That is when `save_top_k` is -1 or >0